metric/system/process/process_darwin_test.go

```diff
@@ -0,0 +1,36 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package process
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestFetchNumThreads only checks some value is returned for NumThreads. However,
+// it cannot verify the correctness of the value.
+func TestFetchNumThreads(t *testing.T) {
```

Comment:
You could test this by writing a program that runs a known number of threads. The trick is that it can't be a Go program to get control over it. A simple C program with a main function that just sleeps until it is terminated is guaranteed to be only one thread, or you could spawn a known number of threads directly with the pthreads API, or you could write it in C++ and use with std:thread which is probably easier to do.

Hardest part would be getting CI to compile the program but this repository already depends on CGO so there must be a C compiler available already.

Reply:
This would also work for Linux, and if you used `std:thread` in C++ it would also work for Windows. As mentioned, the biggest hurdle will be ensuring you have a C++ toolchain available with the appropriate version of C++ available for `std:thread` support but I hope this is already the case.